### PR TITLE
rangeawareportperbroker_plain kubernetes example

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

- Enhancement / new feature

### Description

Kube example illustrating rangeawareportperbroker with a Strimzi controlled Kafka cluster using discontiguous node ids.

### Additional Context

I put this together to see @robobario #1220 work before my eyes.   Having it in the examples may help others.

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

**Comment by robobario** [comment]
Thanks Keith, this will be useful to have as a basis for exploring/testing against different KRaft topologies. I'm wondering if it would be better named along those lines rather than being about the range aware exposition? Like `kubernetes-examples/kraft-cluster`, so it's more about demonstrating integration with an interesting KRaft topology.

**Comment by k-wall** [comment]
> Like kubernetes-examples/kraft-cluster, so it's more about demonstrating integration with an interesting KRaft topology.

Might that name not mislead?  It could be interpreted as **the** way to expose a KRaft cluster, when the user has the option to use SNI too.

I do think that the examples currently munge two ideas together. One is about network exposition and one is about demonstrating the filters.    Maybe dividing them on those lines would be an improvement.

**Comment by SamBarker** [comment]
>I do think that the examples currently munge two ideas together. One is about network exposition and one is about demonstrating the filters. Maybe dividing them on those lines would be an improvement.

I think adding a layer would help yes, but I'm not convinced by `kubernetes-examples/network-exposition` maybe `deployments` or `external-access`. I think I'd lean towards `deployments` as thats where we would put an example of using Kroxylicious  as a side car proxy and eventually the gateway-api. 

I'm fully on board with `kubernetes-examples/filters`

**Comment by k-wall** [comment]
the examples would beneath ./filters would still be deployments, so they exist to highlight a filter thing rather than a networky thing.

**Comment by SamBarker** [comment]
how about `./topologies/` or the more explicit `./network-topologies`  ?

**Comment by k-wall** [comment]
@SamBarker or @robobario  - could you re-review, please merge if happy.

File: kubernetes-examples/network-topologies/rangeawareportperbroker_plain/base/kafka/kafka-persistent.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  replicas: 1
+  roles:
+    - controller
+    - broker
+  storage:
+    type: jbod
+    volumes:
+      - id: 0
+        type: persistent-claim
+        size: 100Gi
```
Comment (SamBarker) [inline kubernetes-examples/network-topologies/rangeawareportperbroker_plain/base/kafka/kafka-persistent.yaml:73]: Thats quite a generous request for a sample...

File: kubernetes-examples/network-topologies/rangeawareportperbroker_plain/base/kafka/kafka-persistent.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      - name: tls
+        port: 9093
+        type: internal
+        tls: true
+    config:
+      offsets.topic.replication.factor: 2
+      transaction.state.log.replication.factor: 2
+      transaction.state.log.min.isr: 2
+      default.replication.factor: 2
+      min.insync.replicas: 2
```
Comment (SamBarker) [inline kubernetes-examples/network-topologies/rangeawareportperbroker_plain/base/kafka/kafka-persistent.yaml:31]: I know its just a demo, but setting min.isr & min.insync to 2 when we only have 2 nodes feels a bit off.

File: kubernetes-examples/network-topologies/rangeawareportperbroker_plain/postinstall.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+echo "Note: kcat 1.7 -P will send each line as a separate record, but you need to press ^D to actually get kcat to start sending - see  https://github.com/edenhill/kcat/issues/322"
+echo "quay.io/kroxylicious/kcat:1.6.0 will send each line as a record on pressing enter."
+
+echo "To produce to kroxylicious:"
+echo "kubectl -n kroxylicious run kcat-kafka-producer -ti --image=${KCAT} --rm=true --restart=Never -- -P -t ${TOPIC} -b ${BOOTSTRAP}"
+echo "To consume from kroxylicious:"
+echo "kubectl -n kroxylicious run kcat-kafka-consumer -ti --image=${KCAT} --rm=true --restart=Never -- -C -t ${TOPIC} -b ${BOOTSTRAP} -o beginning"
+echo "To see the node list exposed by the virtual cluster:"
+echo "kubectl -n kroxylicious run node-list -ti --image=quay.io/kroxylicious/kaf --rm=true --restart=Never -- kaf node ls -b ${BOOTSTRAP}"
+
```
Comment (SamBarker) [inline kubernetes-examples/network-topologies/rangeawareportperbroker_plain/postinstall.sh:38]: Entirely non blocking thought. If we are going to demo both Kafka and Kcat should we also include KAF?

**Review by SamBarker — APPROVED** [review]
LGTM 

@robobario merge if you are happy

**Comment by SamBarker** [comment]
Apparently I started my review on stale view so didn't see @robobario approval already. Have gone ahead and merged.

**Pull request merged**